packages/onboarding/src/AssertUtil.ts

```diff
@@ -0,0 +1,488 @@
+import OnboardingClient from './OnboardingClient'
+import BN from 'bn.js'
+import {
+  DC_MINT,
+  IOT_MINT,
+  MOBILE_MINT,
+  toBN,
+  HNT_MINT,
+  heliumAddressToSolPublicKey,
+} from '@helium/spl-utils'
+import { subDaoKey } from '@helium/helium-sub-daos-sdk'
+import { Connection, PublicKey, AccountInfo, Cluster, LAMPORTS_PER_SOL } from '@solana/web3.js'
+import axios from 'axios'
+import {
+  AccountLayout,
+  createAssociatedTokenAccountIdempotentInstruction,
+  getAccount,
+  getAssociatedTokenAddressSync,
+} from '@solana/spl-token'
+import {
+  iotInfoKey,
+  mobileInfoKey,
+  rewardableEntityConfigKey,
+} from '@helium/helium-entity-manager-sdk'
+import * as Currency from '@helium/currency-utils'
+import {
+  BONES_IN_HNT,
+  DcProgram,
+  FULL_LOCATION_STAKING_FEE,
+  HemProgram,
+  HotspotType,
+  TXN_FEE_IN_LAMPORTS,
+} from './types'
+
+type Account = AccountInfo<string[]>
+
+const getOraclePriceFromSolana = async (opts: { connection: Connection; cluster: Cluster }) => {
+  const price = await Currency.getOraclePrice({ tokenType: 'HNT', ...opts })
+  if (!price?.aggregate.price) {
+    throw new Error('Failed to fetch oracle price')
+  }
+
+  return new BN(price.aggregate.price)
+}
+
+const getBalance = async (wallet: PublicKey, connection: Connection, mint: PublicKey) => {
+  const balance = await Currency.getBalance({
+    pubKey: wallet,
+    connection,
+    mint: new PublicKey(mint),
+  })
+  return new BN(balance.toString())
+}
+
+const getBalances = async (wallet: PublicKey, connection: Connection) => {
+  const sol = await connection.getBalance(wallet)
+  const hnt = (await getBalance(wallet, connection, HNT_MINT)).div(BONES_IN_HNT)
+  const dc = await getBalance(wallet, connection, DC_MINT)
+
+  return { hnt, dc, sol: new BN(sol) }
+}
+
+const burnHNTForDataCredits = async ({
+  dcAmount,
+  wallet,
+  connection,
+  dcProgram,
+}: {
+  dcAmount: BN
+  wallet: PublicKey
+  connection: Connection
+  dcProgram: DcProgram
+}) => {
+  const txn = await dcProgram.methods
+    .mintDataCreditsV0({
+      hntAmount: null,
+      dcAmount,
+    })
+    .preInstructions([
+      createAssociatedTokenAccountIdempotentInstruction(
+        wallet,
+        getAssociatedTokenAddressSync(DC_MINT, wallet, true),
+        wallet,
+        DC_MINT,
+      ),
+    ])
+    .accounts({
+      dcMint: DC_MINT,
+      recipient: wallet,
+    })
+    .transaction()
+
+  if (!txn) return
+
+  const { blockhash } = await connection.getLatestBlockhash('finalized')
+  txn.recentBlockhash = blockhash
+  txn.feePayer = wallet
+
+  return txn
+}
+
+const getAtaAccountCreationFee = async (solanaAddress: PublicKey, connection: Connection) => {
+  const ataAddress = getAssociatedTokenAddressSync(DC_MINT, solanaAddress, true)
+
+  try {
+    await getAccount(connection, ataAddress)
+    return new BN(0)
+  } catch {
+    return new BN(2039280)
```

Comment:
Should use the connection to get the rent cost for the size of the account, as rent may be dynamic in the near to mid future.